vite.config.ts

```diff
@@ -1,40 +1,56 @@
 import react from '@vitejs/plugin-react';
-import { defineConfig } from 'vite';
 import path from 'path';
+import { defineConfig, loadEnv } from 'vite';
 import checker from 'vite-plugin-checker';
 
-export default defineConfig({
-  define: {
-    __APP_VERSION__: JSON.stringify(process.env.npm_package_version),
-  },
-  plugins: [
-    react(),
-    checker({
-      eslint: {
-        files: ['./src'],
-        extensions: ['.tsx', '.ts'],
+export default defineConfig(({ mode }) => {
+  // Ref: https://github.com/vitejs/vite/issues/3105#issuecomment-939703781
+  const env = loadEnv(mode, 'env');
+  const htmlPlugin = () => {
+    return {
+      name: 'html-transform',
+      transformIndexHtml(html: string) {
+        return html.replace(/<%=(.*?)%>/g, function (match, p1) {
+          return env[p1];
+        });
+      },
+    };
+  };
```

Comment:
I swapped this out with this more precise version:

```js
const htmlPlugin = () => ({
    name: "html-transform",
    transformIndexHtml: html =>
      html.replace(
        /<%=\s*([a-zA-Z_]+)\s*%>/g,
        (_match, variableName) => env[variableName]
      )
});
```

Main advantage being that it supports ignoring whitespace inside the tags and only includes `a-zA-Z` and `_` in the allowed list of letters for env variables.